web/src/engine/websites/ToCoronaEx.ts

```diff
@@ -0,0 +1,129 @@
+import DeScramble from '../transformers/ImageDescrambler';
+
+type APIResults<T> = {
+    next_cursor: string,
+    resources: T
+}
+
+type APIManga = {
+    id: string,
+    title: string
+}
+
+type APIChapter = {
+    id: string,
+    title: string,
+    episode_order: number
+}
+
+type APIPages = {
+    pages: {
+        page_image_url: string,
+        drm_hash: string
+    }[]
+}
+export default class extends DecoratableMangaScraper {
+
+    protected apiurl = 'https://api.to-corona-ex.com';
+    protected apikey = 'K4FWy7Iqott9mrw37hDKfZ2gcLOwO-kiLHTwXT8ad1E=';
+
+    public constructor(public readonly Identifier: string = 'to-corona-ex', public readonly Title: string = 'コロナ (to-corona-ex)', url: string = 'https://to-corona-ex.com', tags: Tag[] = [Tags.Language.Japanese, Tags.Media.Manga, Tags.Source.Official]) {
+        super(Identifier, Title, url, ...tags);
+    }
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/comics/\\d+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.match(/\/comics\/(\d+)/)[1];
+        const data = await FetchJSON<APIManga>(this.CreateRequest(new URL(`/comics/${id}`, this.apiurl)));
+        return new Manga(this, provider, data.id, data.title.replace('@COMIC', '').trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let url: URL | null = new URL('/comics?order=asc&sort=title_yomigana', this.apiurl); url != null;) {
+            const { resources, next_cursor } = await FetchJSON<APIResults<APIManga[]>>(this.CreateRequest(url));
+            resources.forEach(manga => mangaList.push(
+                new Manga(this, provider, manga.id, manga.title.replace('@COMIC', '').trim())
+            ));
+            url = next_cursor ? new URL(`/comics?order=asc&sort=title_yomigana&after_than=${next_cursor}`, this.apiurl) : null;
+        }
+        return mangaList.distinct();
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterList: Chapter[] = [];
+
+        for (let url: URL | null = new URL(`/episodes?comic_id=${manga.Identifier}&episode_status=free_viewing%2Conly_for_subscription&order=asc&sort=episode_order`, this.apiurl); url != null;) {
+            const { resources, next_cursor } = await FetchJSON<APIResults<APIChapter[]>>(this.CreateRequest(url));
+            resources.forEach(chapter => chapterList.push(
+                new Chapter(this, manga, chapter.id, chapter.title.trim())
+            ));
+            url = next_cursor ? new URL(`/episodes?comic_id=${manga.Identifier}&episode_status=free_viewing%2Conly_for_subscription&order=asc&sort=episode_order&after_than=${next_cursor}`, this.apiurl) : null;
```

Comment:
Just update the `searchParameters` of the URL, or break loop if no next cursor is available